pkgs/by-name/ko/koboldcpp/package.nix

```
@@ -133,10 +149,26 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     runHook postInstall
   '';
 
+  # `env.NIX_CFLAGS_COMPILE` is not respected by KoboldCpp, so the Makefile has to be manually patched
+  # https://discourse.nixos.org/t/env-nix-cflags-compile-vs-cxxflags/39192/8
+  # https://github.com/LostRuins/koboldcpp/blob/0fc18d2d82633d107a85c5ba725f665c8228e1c7/Makefile#L136
+  # https://github.com/LostRuins/koboldcpp/blob/0fc18d2d82633d107a85c5ba725f665c8228e1c7/Makefile#L73C1-L73C69
+  # Remove `warning: argument unused during compilation: '-s' [-Wunused-command-line-argument]` on Darwin
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-fail '		CFLAGS += -march=native -mtune=native' '		CFLAGS += -march=${
+        (if (march != "") then "${march}" else "native")
+      } -mtune=${(if (mtune != "") then "${mtune}" else "native")}' \
```

Wouldn't this get the job done?

```suggestion
      --replace-fail "-march=native" "" \
      --replace-fail "-mcpu=native" "" \
      --replace-fail "-mtune=native" "" \
```

---

How so? It just removes those flags rather than replacing them with custom one